src/main/java/com/yugabyte/sample/apps/SqlGeoPartitionedTable.java

```diff
@@ -0,0 +1,222 @@
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied.  See the License for the specific language governing permissions and limitations
+// under the License.
+//
+package com.yugabyte.sample.apps;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+
+import com.yugabyte.sample.common.SimpleLoadGenerator.Key;
+
+/**
+ * This workload writes and reads some random string keys from a postgresql table.
+ */
+public class SqlGeoPartitionedTable extends AppBase {
+  private static final Logger LOG = Logger.getLogger(SqlGeoPartitionedTable.class);
+
+  // Static initialization of this workload's config. These are good defaults for getting a decent
+  // read dominated workload on a reasonably powered machine. Exact IOPS will of course vary
+  // depending on the machine and what resources it has to spare.
+  static {
+    // Disable the read-write percentage.
+    appConfig.readIOPSPercentage = -1;
+    // Set the read and write threads to 2 each.
+    appConfig.numReaderThreads = 2;
+    appConfig.numWriterThreads = 2;
+    // The number of keys to read.
+    appConfig.numKeysToRead = -1;
+    // The number of keys to write. This is the combined total number of inserts and updates.
+    appConfig.numKeysToWrite = -1;
+    // The number of unique keys to write. This determines the number of inserts (as opposed to
+    // updates).
+    appConfig.numUniqueKeysToWrite = NUM_UNIQUE_KEYS;
+  }
+
+  // The default table name to create and use for CRUD ops.
+  private static final String DEFAULT_TABLE_NAME = "SqlGeoPartitionedTable";
+
+  // The shared prepared select statement for fetching the data.
+  private volatile PreparedStatement preparedSelect = null;
+
+  // The shared prepared insert statement for inserting the data.
+  private volatile PreparedStatement preparedInsert = null;
+
+  // Lock for initializing prepared statement objects.
+  private static final Object prepareInitLock = new Object();
+
+  public SqlGeoPartitionedTable() {
+    buffer = new byte[appConfig.valueSize];
+  }
+
+  /**
+   * Drop the table created by this app.
+   */
+  @Override
+  public void dropTable() throws Exception {
+    Connection connection = getPostgresConnection();
+    connection.createStatement().execute("DROP TABLE " + getTableName());
+    LOG.info(String.format("Dropped table: %s", getTableName()));
+  }
+
+  @Override
+  public void createTablesIfNeeded(TableOp tableOp) throws Exception {
+
+    try (Connection connection = getPostgresConnection();
+         Statement statement = connection.createStatement()) {
+
+      // (Re)Create the table (every run should start cleanly with an empty table).
```

Comment:
Do we need to start with a clean table all the time? 
Also If you want the start cleanly shouldn't you always drop the table?